### PR TITLE
Upgrade testing user event library 

### LINKED ISSUE #1962: Upgrade to latest `@testing-library/user-event`
This would need some update to the current tests to make it compatible.

@CalvinChiramal _a

### PR DESCRIPTION
**Pull request by ittyjose**
- Fixes #1962 

**Description**
Updated package userEvent to latest for testing.
**Checklist**

~- [ ] I have made corresponding changes to the documentation.~
~- [ ] I have updated the types definition of modified exports.~
~- [ ] I have verified the functionality in some of the neeto web-apps.~
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added the necessary label (patch/minor/major - If package publish
      is required).

**Reviewers**
@ajmaln

### CODE REVIEW

**Comment by ittyjose** [comment]
please review @ajmaln _a

File: tests/formik/BlockNavigation.test.jsx [hunk 1]
```diff
@@ -49,36 +49,36 @@ describe("formik/BlockNavigation", () => {
...
     render(<TestBlockNavigation />);
 
     const input = screen.getByRole("textbox");
-    await waitFor(() => {
-      userEvent.type(input, "test");
-      userEvent.click(screen.getByRole("link"));
+    await waitFor(async () => {
+      await userEvent.type(input, "test");
+      await userEvent.click(screen.getByRole("link"));
     });
```
Comment (ajmaln) [inline tests/formik/BlockNavigation.test.jsx:66]: This `waitFor` might not be required, could you check?

**Review by ajmaln — APPROVED** [review]
@ittyjose _a A minor suggestion, please update and merge

File: tests/formik/BlockNavigation.test.jsx (hunk 1 above)
Comment (ittyjose) [reply tests/formik/BlockNavigation.test.jsx]: You are right @ajmaln. I have updated it accordingly.

**Pull request merged**